editoast/src/views/v2/path/projection.rs

```diff
@@ -0,0 +1,471 @@
+        let mut has_next = false;
+        let mut has_prev = false;
+        let first_track_loc = if found_track_range.direction == Direction::StartToStop {
+            let offset = found_track_range.begin + position - found_position;
+            if offset == found_track_range.begin && index > 0 {
+                has_prev = true;
+            } else if offset == found_track_range.end && index < self.path.len() - 1 {
+                has_next = true;
+            }
+            TrackLocation::new(&found_track_range.track_section, offset)
+        } else {
+            let offset = found_track_range.end - (position - found_position);
+            if offset == found_track_range.end && index > 0 {
+                has_prev = true;
+            } else if offset == found_track_range.begin && index < self.path.len() - 1 {
+                has_next = true;
+            }
+            TrackLocation::new(&found_track_range.track_section, offset)
+        };
+
+        // Position on a single location
+        if !has_prev && !has_next {
+            return TrackLocationFromPath::One(first_track_loc);
+        }
+
+        // Position between two locations
+        let second_track_loc = if has_next {
+            let track_range = &self.path[index + 1];
+            TrackLocation::new(&track_range.track_section, track_range.start())
+        } else {
+            let track_range = &self.path[index - 1];
+            TrackLocation::new(&track_range.track_section, track_range.stop())
+        };
+        TrackLocationFromPath::Two(first_track_loc, second_track_loc)
+    }
+
+    /// Returns a list of intersection ranges between track ranges and the self.
+    /// Intersection ranges are positions on the `track_ranges` path.
+    /// Note: If there is no common track section range, the returned list is empty.
+    pub fn get_intersections(&self, track_ranges: &[TrackRange]) -> Vec<(u64, u64)> {
+        let mut intersection_list = vec![];
+        // Handle the length computation in mm
+        let mut current_pos: u64;
+        let mut next_pos: u64 = 0;
+
+        // Memorizes intersection ranges
+        let mut start_intersection: Option<u64> = None;
+        let mut end_intersection: u64 = 0;
+        // Memorize the index of a track section in a path
+        let mut path_track_index: usize = 0;
+
+        for track_range in track_ranges {
+            // Handle current position
+            current_pos = next_pos;
+            next_pos += track_range.end - track_range.begin;
+
+            // When a track is not part of self
+            let Some(proj_track_range) = self.get_track_range(&track_range.track_section) else {
+                if let Some(start_boundary) = start_intersection {
+                    intersection_list.push((start_boundary, end_intersection));
+                }
+                start_intersection = None;
+                continue;
+            };
+
+            // When a track is in self but not in `track_ranges` path
```

Comment:
I think this comment is misleading. The track is always in `track_ranges` (`for track_range in track_ranges`). I'd suggest:

```suggestion
            // When a previous `track_range` (from maybe several iterations ago) left the `self` path, but the current `track_range` is back on it, then `dist > 1` which indicates a discontinuity in the resulting intersection list. So we need to close the previous intersection (if there is one => `start_intersection.is_some()`).
```